src/data/reducers/auth.js

```diff
@@ -14,21 +14,29 @@ const initialState = {
 
 export default function (state = initialState, action) {
   const { type, payload } = action;
+  let firstName, lastName, email, userId, teamId;
 
   switch (type) {
     case REGISTER_SUCCESS:
+      ({ firstName, lastName, email, userId, teamId } = payload);
       return {
         ...state,
         // TODO: email, userId, firstName, lastName, should be set here
-        user: null,
+        user: {
+          payload,
+          lastName,
+          email,
+          userId,
+          teamId,
+        },
```

Comment:
I think this stores a payload object within the user object and then sets the same fields in payload into the user object again which ends up looking like this:
```
user: {
    payload: {
        firstName: 'asd',
	lastName: 'fgh',
	email: 'asd@gmail.com',
	userId: 123,
	teamId: 456
    },
    lastName: 'fgh',
    email: 'asd@gmail.com',
    userId: 123,
    teamId: 456
}
```

Not sure if I'm understanding this correctly or if the above is what we want but if payload is everything we expect to be stored in the user object, we can pass it directly which ends up looking like this:

```
user: {
    firstName: 'asd',
    lastName: 'fgh',
    email: 'asd@gmail.com',
    userId: 123,
    teamId: 456
}
```

```suggestion
        user: payload
```

Reply:
Good point, thanks!